### ISSUE TITLE
Investigate: subsetting a `MeshCoord` should return a `MeshCoord`?

### ISSUE THREAD

**Issue by trexfeathers** [SciTools/iris#5991]
I _might_ have some code that could help with this:

https://github.com/SciTools/iris/blob/6adac1b6f83baa2e1d4a13ced0d6748ff91092ee/lib/iris/tests/stock/__init__.py#L840-L886

**Comment by trexfeathers** [comment]
This is very closely linked to #4751 - it's probably a choice between 1 and the other

**Comment by trexfeathers** [comment]
The below code demonstrates the currently typical way of recreating a mesh after subsetting (#4751 refers to this as 'remeshing').

If we make the suggestions here work then the below code could create some confusing situations - no `MeshCoord` will have been replaced with an `AuxCoord`. But it will be easy to avoid that happening if we just add a clause into `Mesh.from_coords()` that knows to perform a 'no-op' if the coordinates passed in are already `MeshCoord`s - just return the existing `Mesh`.

If we made that work I would be comfortable implementing changes even after mesh support becomes non-experimental.

https://github.com/SciTools/iris/blob/409f92cf7e77c8c3f7173df10d22f4511b301ed7/lib/iris/experimental/geovista.py#L324-L333
